ethereum/contracts/SuperAdjudicator.sol

```diff
@@ -0,0 +1,147 @@
+uint256 constant MASK_250 = 2**250 - 1;
+uint256 constant ZORRO_SUPER_ADJUDICATE_SELECTOR = uint256(
+  keccak256('super_adjudicate') & bytes32(MASK_250)
+);
+
+interface IStarknetCore {
+  // Sends a message to an L2 contract. Returns the hash of the message.
+  function sendMessageToL2(
+    uint256 to_address,
+    uint256 selector,
+    uint256[] calldata payload
+  ) external returns (bytes32);
+}
+
+// Interface for https://github.com/kleros/arbitrable-proxy-contracts/blob/master/contracts/ArbitrableProxy.sol
+interface IArbitrableProxy {
+  function arbitrator() external returns (IArbitrator);
+
+  function createDispute(
+    bytes calldata _arbitratorExtraData,
+    string calldata _metaevidenceURI,
+    uint256 _numberOfRulingOptions
+  ) external payable returns (uint256 disputeID);
+}
+
+contract SuperAdjudicator {
+  struct ArbitratorConfiguration {
+    bytes arbitratorExtraData;
+    string metaevidenceURI;
+    uint256 numberOfRulingOptions;
+  }
+
+  event ArbitratorConfigurationChanged(
+    bytes arbitratorExtraData,
+    string metaevidenceURI,
+    uint256 numberOfRulingOptions
+  );
+
+  // Set once during construction
+  IStarknetCore public immutable starknetCore;
+  IArbitrableProxy public immutable arbitrableProxy;
+  uint256 public immutable zorroL2Address;
+
+  address public owner; // Owner can modify arbitrator configuration
+  ArbitratorConfiguration public arbitratorConfiguration;
+  mapping(uint256 => uint256) profileIdToDisputeId;
+
+  constructor(
+    IStarknetCore _starknetCore,
+    IArbitrableProxy _arbitrableProxy,
+    uint256 _zorroL2Address,
+    address _owner,
+    bytes memory _arbitratorExtraData,
+    string memory _metaevidenceURI,
+    uint256 _numberOfRulingOptions
+  ) {
+    starknetCore = _starknetCore;
+    arbitrableProxy = _arbitrableProxy;
+    zorroL2Address = _zorroL2Address;
+    owner = _owner;
+    _setPolicy(_arbitratorExtraData, _metaevidenceURI, _numberOfRulingOptions);
+  }
+
+  function setOwner(address newOwner) external {
+    require(msg.sender == owner, 'caller is not the owner');
+    owner = newOwner;
+  }
+
+  function setPolicy(
+    bytes calldata arbitratorExtraData,
+    string calldata metaevidenceURI,
+    uint256 numberOfRulingOptions
+  ) external {
+    require(msg.sender == owner, 'caller is not the owner');
+    _setPolicy(arbitratorExtraData, metaevidenceURI, numberOfRulingOptions);
+  }
+
+  function _setPolicy(
+    bytes calldata arbitratorExtraData,
+    string calldata metaevidenceURI,
+    uint256 numberOfRulingOptions
+  ) internal {
+    arbitratorConfiguration.arbitratorExtraData = arbitratorExtraData;
+    arbitratorConfiguration.metaevidenceURI = metaevidenceURI;
+    arbitratorConfiguration.numberOfRulingOptions = numberOfRulingOptions;
+    emit ArbitratorConfigurationChanged(
+      arbitratorExtraData,
+      metaevidenceURI,
+      numberOfRulingOptions
+    );
+  }
+
+  function createDispute(uint256 profileId) external {
```

Comment:
It looks like we don't have any StarkNet->Ethereum communication here, so I assume we're just depending on the policy to say something to the effect of "if an appeal comes in and the profile isn't in an appealable state, automatically deny the appeal"?

Reply:
(for posterity, yup that's the intention — will likely have a third kleros ruling option for "spurious case")